chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -918,6 +958,8 @@ impl<'a, S: BlockchainStorageWrite, V: TransactionVerificationStrategy> Chainsta
     ) -> Result<GenBlockIndex, BlockError> {
         let best_block_id = self
             .get_best_block_id()
+            // FIXME: here we actually "expect" on a db error, not on the absence of the best block
```

Comment:
Disconnection should never fail. There's absolutely no way to recover from it.

Think about it: We're basically unable to switch to another chain due to I/O error. What are we supposed to do, exactly?!